src/speed-index.js

```diff
@@ -33,14 +33,44 @@ function calculateFrameProgress(current, initial, target) {
 	return progress;
 }
 
-function calculateVisualProgress(frames) {
+function calculateProgressBetweenFrames(frames, lowerBound, upperBound, threshold, getProgress, setProgress) {
+	if (threshold === 0) {
+		frames.forEach(frame => setProgress(frame, getProgress(frame)));
+		return;
+	}
+
+	const lowerFrame = frames[lowerBound];
+	const upperFrame = frames[upperBound];
+
+	const lowerProgress = getProgress(lowerFrame);
+	const upperProgress = getProgress(upperFrame);
+
+	setProgress(lowerFrame, lowerProgress);
+	setProgress(upperFrame, upperProgress);
+
+	if (Math.abs(lowerProgress - upperProgress) < threshold) {
+		for (let i = lowerBound; i < upperBound; i++) {
+			setProgress(frames[i], lowerProgress);
```

Comment:
should we also set a property on the frame indicating this value was interpolated? just an idea.